sgxs-loaders/src/win_sgx/mod.rs

```diff
@@ -0,0 +1,342 @@
+                flags.insert(WinPageProperties::PAGE_READWRITE);
+            }
+            else if eadd.secinfo.flags.contains(SecinfoFlags::X | SecinfoFlags::R) {
+                flags.insert(WinPageProperties::PAGE_EXECUTE_READ);
+            }
+            else if eadd.secinfo.flags.contains(SecinfoFlags::W | SecinfoFlags::X) {
+                return Err(Error::Add(LibraryError::InvalidParameter));
+            }
+            else if eadd.secinfo.flags.contains(SecinfoFlags::R) {
+                flags.insert(WinPageProperties::PAGE_READONLY);
+            }
+            else if eadd.secinfo.flags.contains(SecinfoFlags::X) {
+                flags.insert(WinPageProperties::PAGE_EXECUTE);
+            }
+            else {
+                return Err(Error::Add(LibraryError::InvalidParameter));
+            }
+        }
+
+        match PageType::from_repr(eadd.secinfo.flags.page_type()) {
+            Some(PageType::Reg) => {}
+            Some(PageType::Tcs) => flags.insert(WinPageProperties::PAGE_TCS),
+            _ => return Err(Error::Add(LibraryError::InvalidParameter)),
+        }
+        match chunks.0 {
+            0 => flags.insert(WinPageProperties::PAGE_UNVALIDATED),
+            0xffff => {}
+            _ => return Err(Error::Add(LibraryError::InvalidParameter)),
+        }
+
+        if flags.contains(WinPageProperties::PAGE_TCS) {
+            // NOTE: For some reason the windows API needs the Read flag set but then removes it
+            assert_eq!(eadd.secinfo.flags.contains(SecinfoFlags::R | SecinfoFlags::W | SecinfoFlags::X), false);
+            flags.insert(WinPageProperties::PAGE_READONLY);
+        }
+        unsafe {
+            let mut error = 0;
+            let mut dataLoaded : usize = 0;
+            let pDataLoaded: *mut usize = &mut dataLoaded;
+            let curhandle: HANDLE = GetCurrentProcess();
+            let ret = LoadEnclaveData(
+                curhandle,
+                (mapping.base + eadd.offset) as _,
+
+                data.as_ptr() as  *const c_void,
+                0x1000,
+                flags.bits(),
+                std::ptr::null(),
+                0,
+                pDataLoaded,
+                &mut error
+            ) ;
+            if ret == 0 {
+                return Err(Error::Add(LibraryError::OS(IoError::last_os_error())));
+            }
+            assert_eq!(dataLoaded, 0x1000);
+        }
+        Ok(())
+    }
+
+    fn init(
+        mapping: &Mapping<Self>,
+        sigstruct: &Sigstruct,
+        einittoken: Option<&Einittoken>,
+    ) -> Result<(), Self::Error> {
+        let mut initInfo : ENCLAVE_INIT_INFO_SGX = ENCLAVE_INIT_INFO_SGX {
+            SigStruct : [0 ; 1808],
+            Reserved1 : [0 ; 240],
+            EInitToken: [0 ; 304],
+            Reserved2 : [0 ; 1744],
+        };
+        initInfo.SigStruct.clone_from_slice(&sigstruct.as_ref());
+        if einittoken.is_some() {
```

Comment:
`is_some` followed by `unwrap` is an anti-pattern. Whenever you see this, use pattern matching instead:
```rust
if let Some(einittoken) = einittoken {
    // use einittoken
}
```